core.d.ts

```diff
@@ -318,6 +318,42 @@ export type ReadableStreamWithFileType = ReadableStream & {
 	readonly fileType?: FileTypeResult;
 };
 
+/**
+Function that allows specifying custom detection mechanisms.
+
+An iterable of detectors can be provided as argument for file type detection options.
+
+The detectors are called before the default detections in the provided order.
+
+Custom detectors can be used to add new FileTypeResults or to modify return behaviour of existing FileTypeResult detections.
+
+If the detector returns `undefined`, the `tokenizer.position` should be 0 (unless it's a stream). That allows other detectors to parse the file.
+
+Example detector array which can be extended and provided to each public method via the fileTypeOptions argument:
+
+	const customDetectors = [
+		async tokenizer => {
+			const unicornHeader = [85, 78, 73, 67, 79, 82, 78]; // "UNICORN" as decimal string
+			const buffer = Buffer.alloc(7);
+			await tokenizer.peekBuffer(buffer, {length: unicornHeader.length, mayBeLess: true});
+			if (unicornHeader.every((value, index) => value === buffer[index])) {
+				return {ext: 'unicorn', mime: 'application/unicorn'};
+			}
+
+			return undefined;
```

Comment:
I just realized, to give the custom scanners the same power as the internal scanner, you may want to be able to distinct the following cases:
1. Scanner does recognize the file, it did not read from the tokenizer, proceed with the next scanner
2. Scanner does kind if recognize the file, but the file exact type could not be determined, hence it also returns undefined. Possibly the scanner went in "level 2 scaniing" and we (read) consumed the data.

Examples of option 2:
- https://github.com/sindresorhus/file-type/blob/ca7a03c26761444e024764a38d16597455281cfd/core.js#L1267
- https://github.com/sindresorhus/file-type/blob/ca7a03c26761444e024764a38d16597455281cfd/core.js#L1483

Reply:
Please note my reply [above](https://github.com/sindresorhus/file-type/pull/603#discussion_r1361559226).